### PR TITLE
`prevent-abbreviations` - non-ASCII characters ignored in filenames

### PR DESCRIPTION
**Pull request by MichaelBlm**
Fixes https://github.com/sindresorhus/eslint-plugin-unicorn/issues/2292

### CODE REVIEW

**Comment by MichaelBlm** [comment]
Here is an explanation of the updated regular expression:

1. `(?=\P{Ll})`: This is a positive lookahead assertion. It matches a position where the next character is not a Unicode lowercase letter. `\P{Ll}` is a Unicode character class that negates the `Ll` (lowercase letter) class.

2. `(?<=\P{L})`: This is a positive lookbehind assertion. It matches a position where the previous character is not a Unicode letter (uppercase or lowercase). `\P{L}` negates the `L` (letter) Unicode character class.

**Comment by fisker** [comment]
The solution looks tricky to me, I prefer just ignore if non-ascii characters included.

File: rules/prevent-abbreviations.js [hunk 1]
```diff
@@ -116,7 +116,7 @@ const getNameReplacements = (name, options, limit = 3) => {
 	}
 
 	// Split words
-	const words = name.split(/(?=[^a-z])|(?<=[^A-Za-z])/).filter(Boolean);
+	const words = name.split(/(?=\P{Ll})|(?<=\P{L})/u).filter(Boolean);
```
Comment (sindresorhus) [inline rules/prevent-abbreviations.js:119]: Use the verbose names (`Lowercase_Letter`), for readability.

**Comment by sindresorhus** [comment]
@fisker Tricky how? The solution looks fine to me.

File: rules/prevent-abbreviations.js (hunk 1 above)
Comment (MichaelBlm) [reply rules/prevent-abbreviations.js]: Updated to include the verbose names

**Comment by MichaelBlm** [comment]
My explanation of the regex wasn't that clear. \P{Letter} is equivalent to [^a-zA-Z] but it includes non-ASCII characters

**Comment by fisker** [comment]
> @fisker Tricky how? The solution looks fine to me.

In different languages, the latin letters can be used with other characters together to be a word, so the non-letter characters not a word boundary.

For example, in Chinese, "B超" "T恤" are words. Similar in Japanese, I think.

File: rules/prevent-abbreviations.js (hunk 1 above)
Comment (fisker) [reply rules/prevent-abbreviations.js]: We should make a rule for it 😄

Comment (fisker) [reply rules/prevent-abbreviations.js]: https://github.com/ota-meshi/eslint-plugin-regexp/issues/720

Comment (MichaelBlm) [reply rules/prevent-abbreviations.js]: I'll work on it actually

**Comment by MichaelBlm** [comment]
> > @fisker Tricky how? The solution looks fine to me.
> 
> In different languages, the latin letters can be used with other characters together to be a word, so the non-letter characters not a word boundary.
> 
> For example, in Chinese, "B超" "T恤" are words. Similar in Japanese, I think.

Can you please provide me a valid/invalid test case to see if this fix doesn't already account for that?

File: rules/prevent-abbreviations.js (hunk 1 above)
Comment (fisker) [reply rules/prevent-abbreviations.js]: Note: the proposal is on another repo.

**Comment by fisker** [comment]
I don't really have a case now.

**Review by fisker — APPROVED** [review]

**Comment by fregante** [comment]
@fisker you're saying that `龖.js` will be counted as "one-letter long", but in Chinese it's a fully-formed "word", not an abbreviation. Hence "ignore if non-ascii characters included"

**Comment by fisker** [comment]
I don't think Chinese characters are "letter"s, but I can be wrong.

**Comment by fregante** [comment]
That's right, it's why quoted "one-letter word". The rule counts characters as "letters" but not all writing systems have letters. 

https://en.wikipedia.org/wiki/Writing_system#Basic_terminology

**Comment by fregante** [comment]
Now that I think about it, we don't need a regex to split words: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Segmenter

**Comment by MichaelBlm** [comment]
I'll explore a solution using the segmenter. Looks promising

**Comment by MichaelBlm** [comment]
I added this code to the valid tests and it is passing with the current implementation? Is that the correct behavior?
`		{
			code: 'foo();',
			filename: '龖.js',
		},`

**Pull request merged**